### DISCUSSION TITLE
How does sort by color work?

### DISCUSSION THREAD

**Discussion by navinthomas** [iSoron/uhabits#1343] [General Discussion] [answered]
I want to know how sort by color works. What value corresponding to color does the app use to sort the colored habits?

**Comment by iSoron** [comment, marked as answer]
Hi @navinthomas, sorting by color uses the same sequence that you can see in the color palette:

> red > yellow > green > blue > purple > brown > grey

**Reply by navinthomas** [reply]
So, it's following a zigzag pattern on the colour palette? 
```
01-02-03-04-05
            |
10-09-08-07-06
|
11-12-13-14-15
            |
20-19-18-17-16
|
21-22-23-24-15
```

**Reply by hiqua** [reply]
The comparator used is there: https://github.com/iSoron/uhabits/blob/dev/uhabits-core/src/jvmMain/java/org/isoron/uhabits/core/models/PaletteColor.kt

IIUC the index in the array gives the order.
